Support ZED in SCS 2 log visualizer and provide scrubber API

calvertdw opened the pull request.

calvertdw commented on scs2-session-visualizer-jfx/src/main/java/us/ihmc/scs2/sessionVisualizer/jfx/session/log/TimestampScrubber.java:
```diff
@@ -154,10 +192,9 @@ private int searchRobotTimestampsForIndex(long queryRobotTimestamp)
       {
-         int nextIndex = -index - 1; // insertionPoint
-         index = nextIndex;
+         index = -index - 1; // the point at which the key would be inserted into the array
```
No change here, just I was like wtf, and added an explanation

PotatoPeeler3000 replied:
Can we leave this? The insertion point is talked about in the documentation

calvertdw commented on scs2-session-logger/build.gradle.kts:
```diff
@@ -21,4 +21,5 @@ mainDependencies {
 }
 
 testDependencies {
+   api("org.apache.commons:commons-math:2.2")
```
Needed for because I moved TimestampScrubberTest

TomaszTB commented on scs2-session-logger/src/main/java/us/ihmc/scs2/session/log/ZEDSVOScrubber.java:
```diff
@@ -0,0 +1,329 @@
+
+      if (leftColorImageSlMatPointer != null && !leftColorImageSlMatPointer.isNull())
+      {
+         sl_mat_free(leftColorImageSlMatPointer, SL_MEM_GPU);
```
```suggestion
         sl_mat_free(leftColorImageSlMatPointer, SL_MEM_CPU);
```

TomaszTB commented on scs2-session-visualizer-jfx/src/main/java/us/ihmc/scs2/sessionVisualizer/jfx/session/log/VideoDataReader.java:
```diff
@@ -18,7 +18,7 @@ public interface VideoDataReader
    void readVideoFrame(long timestamp);
 
-   void cropVideo(File file, File file2, long test, long what, ProgressConsumer heck) throws IOException;
+   void cropVideo(File outputFile, File timestampFile, long startTimestamp, long endTimestamphat, ProgressConsumer monitor) throws IOException;
```
Hmm I kinda liked the original names :rofl: 
```suggestion
   void cropVideo(File outputFile, File timestampFile, long startTimestamp, long endTimestamp, ProgressConsumer monitor) throws IOException;
```

TomaszTB commented on scs2-session-visualizer-jfx/src/main/java/us/ihmc/scs2/sessionVisualizer/jfx/session/log/ZEDSVOVideoDataReader.java:
```diff
@@ -0,0 +1,129 @@
+
+      Pointer leftColorImageSlMatPointer = zedScrubber.getLeftColorImageSlMatPointer();
+
+      Mat imageMat = new Mat(imageHeight, imageWidth, opencv_core.CV_8UC(4), // BGRA8
```
```suggestion
      Mat imageMat = new Mat(imageHeight, imageWidth, opencv_core.CV_8UC4, // BGRA8
```

TomaszTB reviewed (commented):
~~This will require a release of ihmc-robot-data-logger, right? Since it uses the new zed-java-api release~~

Forgot there was also a PR in the logger repo

calvertdw commented on scs2-session-logger/src/main/java/us/ihmc/scs2/session/log/BlackMagicScrubber.java:
I moved this stuff to scs2-session-logger so it can be used by RDX without the JavaFX stuff

calvertdw committed:
Add delay compensation.
Pass ZED FPS around.
Optimize SVO frame grabbing further.
Work on writing video files and getting the threading & performance right.
Add coordinate system init parameters.
Get cameraID so sl API can be used.
Get RDX SCS 2 SVO scrubbing working pretty well.
Add error message when svos exist but ZED SDK is not installed.
Cropping SVOs working.
Extract BlackMagicScrubber and move VideoConverter so they can be used in non JavaFX apps.
Extract MagewellScrubber so it can be used in non JavaFX apps.
Extract ZEDSVOScrubber so it can be used with non JavaFX apps.
Import TimestampScrubber.
Move TimestampScrubber so it can be used in non JavaFX apps.

calvertdw committed:
Update to new logger API

calvertdw committed:
Add documentation.

ds58 commented on scs2-session-logger/src/main/java/us/ihmc/scs2/session/log/BlackMagicScrubber.java:
```diff
@@ -0,0 +1,129 @@
+import java.io.PrintWriter;
+
+/**
+ * Provides support for scurbbing images from .mov files recorded from the Decklink cards.
```
typo "scurbbing"

TomaszTB commented on scs2-session-logger/src/main/java/us/ihmc/scs2/session/log/MagewellScrubber.java:
```diff
@@ -0,0 +1,170 @@
+import java.io.PrintWriter;
+
+/**
+ * Provides support for scurbbing images from .mov files recorded with the Magewell logger.
```
another typo "scurbbing"

calvertdw committed:
Fix thing

calvertdw committed:
Fix typos.
